contracts/blockchainTest.scrypt

```diff
@@ -3,36 +3,17 @@ import "blockchain.scrypt";
 
 contract BlockchainTest {
     
-    int blockchainTarget;
-    
-    public function testBlockHeight(BlockHeader bh, MerklePaths merklePath,  bytes coinbaseTx, int height, SigHashPreimage txPreimage) {
-        require(Tx.checkPreimage(txPreimage));
-
-        // get id of previous tx
-        Sha256 prevTxid = Sha256(Util.outpoint(txPreimage)[:32]);
-
-        // validate block header
-        require(Blockchain.isBlockHeaderValid(bh, this.blockchainTarget));
-
-        // verify previous tx is in the block
-        require(Blockchain.txInBlock(prevTxid, bh, merklePath));
-
-        require(Blockchain.blockHeight(bh, coinbaseTx, merklePath)  == height);
+    public function testBlockHeight(BlockHeader bh, MerkleProof merkleproof,  bytes coinbaseTx, int expectedHeight) {
+        require(Blockchain.blockHeight(bh, coinbaseTx, merkleproof)  == expectedHeight);
     }
 
+    public function testBlockTxCount(BlockHeader bh, MerkleProof merkleproofOfLastTx, Sha256 lastTxid, int expectedCount) {
+        require(Blockchain.blockTxCount(bh, lastTxid, merkleproofOfLastTx)  == expectedCount);
+    }
 
-    public function testBlockTxCount(BlockHeader bh, MerklePaths merklePath, MerklePaths merklePathOfLastTx, Sha256 lastTxid, int count, SigHashPreimage txPreimage) {
-        require(Tx.checkPreimage(txPreimage));
-
-        // get id of previous tx
-        Sha256 prevTxid = Sha256(Util.outpoint(txPreimage)[:32]);
-
-        // validate block header
-        require(Blockchain.isBlockHeaderValid(bh, this.blockchainTarget));
-
-        // verify previous tx is in the block
-        require(Blockchain.txInBlock(prevTxid, bh, merklePath));
-
-        require(Blockchain.blockTxCount(bh, lastTxid, merklePathOfLastTx)  == count);
+    public function testIsBlockHeaderValid(BlockHeader bh,  int blockchainTarget) {
+        require(Blockchain.isBlockHeaderValid(bh, blockchainTarget));
     }
+
+    
```

Comment:
Add a case to test a chain of headers: 1) they are actually chained one after another; 2) NOT chained